pkg/platform/model/buildplanner.go

```diff
@@ -272,24 +295,45 @@ func (bp *BuildPlanner) StageCommit(params StageCommitParams) (strfmt.UUID, erro
 	}
 
 	if params.PackageVersion != "" {
-		requirement.VersionRequirement = []bpModel.VersionRequirement{{bpModel.ComparatorEQ: params.PackageVersion}}
+		requirement.VersionRequirement = []bpModel.VersionRequirement{{bpModel.VersionRequirementComparatorKey: bpModel.ComparatorEQ, bpModel.VersionRequirementVersionKey: params.PackageVersion}}
 	}
 
-	err = script.Update(params.Operation, requirement, *params.TimeStamp)
+	err = expression.Update(params.Operation, requirement, *params.TimeStamp)
 	if err != nil {
 		return "", errs.Wrap(err, "Failed to update build graph")
 	}
 
 	// With the updated build expression call the stage commit mutation
-	request := request.StageCommit(params.Owner, params.Project, params.ParentCommit, script)
+	request := request.StageCommit(params.Owner, params.Project, params.ParentCommit, expression)
 	resp := &bpModel.StageCommitResult{}
 	err = bp.client.Run(request, resp)
 	if err != nil {
-		return "", errs.Wrap(err, "failed to fetch build plan")
+		return "", locale.WrapError(err, "err_buildplanner_stage_commit", "Failed to stage commit, error: {{.V0}}", err.Error())
 	}
 
-	if resp.NotFoundError != nil {
-		return "", errs.New("Commit not found: %s", resp.NotFoundError.Message)
+	if resp.Commit == nil {
+		return "", errs.New("Staged commit is nil")
+	}
+
+	if resp.Commit.Build == nil {
+		if resp.Error != nil {
+			return "", errs.New("Commit not found: %s", resp.Error.Message)
```

Comment:
Do we actually know that this is a "not found" error?

Reply:
Not anymore, there are two error types: `Error` and `NotFoundError`. They both have a `message` field. I'm using a generic `Error` type to try and unmarshal both if we encounter them.